src/partr.c

```diff
@@ -378,8 +379,10 @@ JL_DLLEXPORT void jl_wakeup_thread(int16_t tid)
     JULIA_DEBUG_SLEEPWAKE( wakeup_enter = cycleclock() );
     if (tid == self || tid == -1) {
         // we're already awake, but make sure we'll exit uv_run
-        if (jl_atomic_load_relaxed(&ptls->sleep_check_state) == sleeping)
+        if (jl_atomic_load_relaxed(&ptls->sleep_check_state) == sleeping) {
             jl_atomic_store(&ptls->sleep_check_state, not_sleeping);
+            JL_PROBE_RT_SLEEP_CHECK_UNLOCK(ptls);
+        }
```

Comment:
@vtjnash @kpamnany Is it correct to record this as an unlock event? It seems to be fine reading `jl_task_get_next` but I'm not even sure when it's possible to hit this code path.  (This code path seems to indicate that a task can be in Julia code while in `sleeping` state but I don't see how it happens by looking at `jl_task_get_next`)

Reply:
I'd prefer to disambiguate this transition to `not_sleeping` from the one in `jl_task_get_next()`.  Also I'm not sure why we're calling these 'lock' and 'unlock'. I'd call this one `SLEEP_CHECK_WAKEUP` and the one above `SLEEP_CHECK_WAKE` (since (alas!) we have such function names).

It's true that the thread sleep/wakeup logic needs more review; these names can be revisited when that's refactored.

Reply:
Sounds good to me.

Reply:
@kpamnany can you point me to where `SLEEP_CHECK_WAKE` should go?

Reply:
[This one](https://github.com/JuliaLang/julia/pull/43453/files#diff-12cc999d04343402aae8088cc67f9b6680cc9403d967e400e5235415e0040f47R356) (it's in `wake_thread()`).

Reply:
Ok let me know how this looks. We still have one `SLEEP_CHECK_LOCK`; I imagine you would want that renamed as well?